### ISSUE TITLE
pprz_svd_float function unexpected row reference 

### ISSUE THREAD

**Issue by pakanhai** [paparazzi/paparazzi#2789]
In "math/pprz_matrix_decomp_float.c", the pprz_svd_float function returns a segmentation fault with the inputs:
function:`` int pprz_svd_float(float **a, float *w, float **v, int m, int n)``
inputs: ``(float _svd_U[4][6],  float _svd_W[6],float _svd_V[6][6],4,6)``
The error occurs at line 303
```
...  
/* Accumulation of left-hand transformations. */
  for (i = n - 1; i >= 0; --i) {
    l = i + 1;
    G = w[i];
    if (i < (n - 1)) {
      for (j = l; j < n; ++j) {
        a[i][j] = 0.0;                            <-- HERE
      }
    }
   ...
  }
...
```
It seems the script is trying to access an element that does not exist. Since `m=4` and `n=6` and `a[4][6]`, the first iteration of `a[i][j]` will be `a[5][j]` which does not exist. I think `n` should be replaced with `m` at the start of the loop, this does not give me a segmentation fault and also produces the correct decomposition. 
@gautierhattenberger @dewagter

**Comment by gautierhattenberger** [comment]
@pakanhai I checked the original sources, and it seems they have corrected this with a `min(n,m)` at the start of the for loop, which would be coherent with your segfault and my previous tests working. Could you test that ?

**Comment by gautierhattenberger** [comment]
ok, then if the fix with a `min` works, can you make a pull request for that ? otherwise I will fix it later when I found some time.

thanks for catching this bug

**Comment by gautierhattenberger** [comment]
closed with #2790
